src/mempool.cpp

```diff
@@ -0,0 +1,96 @@
+#include "veriblock/mempool.hpp"
+
+#include "veriblock/stateless_validation.hpp"
+
+namespace altintegration {
+
+bool MemPool::submitATV(const std::vector<ATV>& atvs,
+                        const AltTree& tree,
+                        ValidationState& state) {
+  for (size_t i = 0; i < atvs.size(); ++i) {
+    if (!checkATV(atvs[i], state, tree.getParams(), tree.vbk().getParams())) {
+      return state.Invalid("mempool-submit-atv", i);
+    }
+
+    stored_atvs.push_back(atvs[i]);
+  }
+
+  return true;
+}
+
+bool MemPool::submitVTB(const std::vector<VTB>& vtbs,
+                        const AltTree& tree,
+                        ValidationState& state) {
+  for (size_t i = 0; i < vtbs.size(); ++i) {
+    if (!checkVTB(
+            vtbs[i], state, tree.vbk().getParams(), tree.btc().getParams())) {
+      state.Invalid("mempool-submit-vtb", i);
```

Comment:
Are you sure you want to continue with invalid data? I think you should return state.Invalid here.

Reply:
fixed.